src/command_modules/azure-cli-cognitiveservices/azure/cli/command_modules/cognitiveservices/_params.py

```diff
@@ -3,17 +3,53 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+import re
+import json
 from knack.arguments import CLIArgumentType
+from knack.log import get_logger
 
 from azure.cli.core.commands.parameters import (
     tags_type,
     resource_group_name_type,
     get_resource_name_completion_list)
 
+from azure.cli.core.commands.validators import validate_key_value_pairs
 
+logger = get_logger(__name__)
 name_arg_type = CLIArgumentType(options_list=['--name', '-n'], metavar='NAME')
 
 
+def validate_api_properties(string):
+    """ Extracts JSON format or 'a=b;c=d' format as api properties """
+    try:
+        return json.loads(string)
+    except ValueError:
+        result = validate_key_value_pairs(string)
+        if _is_suspected_json(string):
+            logger.warning('Api properties looks like a JSON format but not valid, interpreted as key=value pairs: %s',
+                           str(result))
+        return result
+
+
+def _is_suspected_json(string):
+    """ If the string looks like a JSON """
+    if string.startswith('{') or string.startswith('\'{') or string.startswith('\"{'):
+        return True
+    if string.startswith('[') or string.startswith('\'[') or string.startswith('\"['):
+        return True
+    if re.match(r"^['\"\s]*{.+}|\[.+\]['\"\s]*$", string):
+        return True
+
+    return False
+
+
+api_properties_type = CLIArgumentType(
+    type=validate_api_properties,
+    help="Api properties in JSON format or a=b;c=d format. Some cognitive services (i.e. QnA Maker) "
```

Comment:
The command-line format should be like tags or metadata: i.e `a=b c=d`. Space-separated key values pairs, not comma or semi-colon separated. You would basically use a validator that looks like this (pulled from Network):

```Python
def validate_metadata(namespace):
    if namespace.metadata:
        namespace.metadata = dict(x.split('=', 1) for x in namespace.metadata)
```